course6/toyapp-adam/displaylib/mesh_util.cc

```diff
@@ -0,0 +1,65 @@
+#include "mesh_util.h"
+
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+
+namespace Display {
+
+static unsigned GetFaceIndex(unsigned index, unsigned vertexCount)
+{
+	return index > 0 ? index - 1 : vertexCount + index;
+}
+
+Mesh CreateMeshFromObjFile(const std::string& filePath)
+{
+	Mesh mesh;
+
+	std::ifstream file(filePath);
+	if (file.fail())
+		return mesh;
+
+	std::vector<Math::Vector3D> vertices;
+
+	std::string line;
+	while (std::getline(file, line))
+	{
+		if (line.length() < 2)
+			continue;
+
+		std::stringstream stream(line);
+		if (line[0] == 'v')
```

Comment:
why don't you just read ```char``` ot ```std::string``` from the stream? IT would be more robust, and you wouldn't have to call ignore()